pkg/schedule/checker/rule_checker.go

```diff
@@ -455,11 +458,47 @@ loopFits:
 			}
 		}
 	}
+
 	// If hasUnhealthyFit is false, it is safe to delete the OrphanPeer.
 	if !hasUnhealthyFit {
 		ruleCheckerRemoveOrphanPeerCounter.Inc()
 		return operator.CreateRemovePeerOperator("remove-orphan-peer", c.cluster, 0, region, fit.OrphanPeers[0].StoreId)
 	}
+
+	// try to use orphan peers to replace unhealthy down peers.
+	for _, orphanPeer := range fit.OrphanPeers {
+		if isUnhealthyPeer(orphanPeer.GetId()) {
+			continue
+		}
+		if pinDownPeer != nil {
+			// no consider witness in this path.
+			if pinDownPeer.GetPeer().GetIsWitness() || orphanPeer.GetIsWitness() {
+				continue
+			}
+			// store should be down.
+			if !c.isStoreDownTimeHitMaxDownTime(pinDownPeer.GetPeer().GetStoreId()) {
+				continue
+			}
+			// check if down peer can replace with orphan peer.
+
+			dstStore := c.cluster.GetStore(orphanPeer.GetStoreId())
+			if fit.Replace(pinDownPeer.GetPeer().GetStoreId(), dstStore) {
```

Comment:
waht it will do if the orphan peer is also down peer?

Reply:
we will skip in the before.